src/borg/archiver.py

```diff
@@ -433,6 +435,14 @@ def item_is_hardlink_master(item):
             return (partial_extract and stat.S_ISREG(item.mode) and
                     item.get('hardlink_master', True) and 'source' not in item)
 
+        if progress:
+            extracted_size = 0
+            for item in archive.iter_items(filter=lambda item: matcher.match(item.path)):
+                extracted_size += item.file_size()
```

Comment:
That's the reason why there's no progress *predictor* here (like windows copy file dialog ;)

![](https://imgs.xkcd.com/comics/estimation.png)

Stuff like unstable / varying internet speeds (cable internet.........) aside I don't think it's too big of an issue.